biocypher/_logger.py

```diff
@@ -60,35 +60,43 @@ def get_logger(name: str = "biocypher") -> logging.Logger:
         now = datetime.now()
         date_time = now.strftime("%Y%m%d-%H%M%S")
 
-        logdir = (
-            _config.config("biocypher").get("log_directory") or "biocypher-log"
-        )
-        os.makedirs(logdir, exist_ok=True)
-        logfile = os.path.join(logdir, f"biocypher-{date_time}.log")
+        log_to_disk = _config.config("biocypher").get("log_to_disk", True)
+
+        if log_to_disk:
+            logdir = (
+                _config.config("biocypher").get("log_directory")
+                or "biocypher-log"
+            )
+            os.makedirs(logdir, exist_ok=True)
+            logfile = os.path.join(logdir, f"biocypher-{date_time}.log")
+
+            # file handler
+            file_handler = logging.FileHandler(logfile)
+
+            if _config.config("biocypher").get("debug"):
+                file_handler.setLevel(logging.DEBUG)
+            else:
+                file_handler.setLevel(logging.INFO)
+
+            file_handler.setFormatter(file_formatter)
+
+            logger.addHandler(file_handler)
 
         # handlers
         # stream handler
         stdout_handler = logging.StreamHandler()
         stdout_handler.setLevel(logging.INFO)
         stdout_handler.setFormatter(stdout_formatter)
 
-        # file handler
-        file_handler = logging.FileHandler(logfile)
-
-        if _config.config("biocypher").get("debug"):
-            file_handler.setLevel(logging.DEBUG)
-        else:
-            file_handler.setLevel(logging.INFO)
-
-        file_handler.setFormatter(file_formatter)
-
         # add handlers
-        logger.addHandler(file_handler)
         logger.addHandler(stdout_handler)
 
         # startup message
         logger.info(f"This is BioCypher v{__version__}.")
-        logger.info(f"Logging into `{logfile}`.")
+        if log_to_disk:
+            logger.info(f"Logging into `{logfile}`.")
+        else:
+            logger.info("Logging into stdout.")
```

Comment:
@nilskre this means that logging into stdout will always only be on the info level and above, never debug, right?

should we inform users that they need to log to disk if they want debug / is that even a good idea? what about passing the log to upstream packages (ie something imports biocypher and then wants access to the debug logs)?

Reply:
Yes (if I understand the code correctly, this is equal to the behavior before in [_logger.py](https://github.com/biocypher/biocypher/blob/main/biocypher/_logger.py#L72), where the stdout_handler is set to INFO and the file_handler is set conditionally.
I agree, that the log level should be the same no matter if it is logged to disk or to stdout.

Reply:
The main question is how to set a debug level log for the file (or some upstream package that may want to collect all debug logs) without spamming the console with detailed debug logs. It should definitely stay the default solution to only print info and above to the console.

Reply:
With the newest commit, we can't have a debug log to file but higher-level console output. I think that is not practical. However, I am not sure I want two log level settings in the config...

Reply:
Ok, I have misunderstood it and consequently removed the last commit.

Reply:
So the current behaviour is to only print INFO and above to the console, except if the user sets `log_to_disk: true` in the config? So it is deactivated by default?

What happens if you set `log-directory` to some value, but not `log_to_disk: true`?

Reply:
Hm, how is the default setting 'true' implemented? The line in the standard config is commented out. If we want it to be true by default, I would just set it there, to be overwritten by the user in case they don't want it.

Reply:
So if we uncomment the standard in the config, we would not need this default=True setting. I think that would be more transparent. I'll push some changes.